src/main/scala/ie/zalando/fabric/gateway/web/marshalling/JsonModels.scala

```diff
@@ -42,8 +43,34 @@ trait JsonModels {
       svcPort <- c.downField("servicePort").as[Option[String]]
     } yield
       FabricServiceDefinition(host.getOrElse("IN-PROGRESS"),
-                              svcName.getOrElse("IN-PROGRESS"),
-                              svcPort.getOrElse(HttpModels.DefaultIngressServiceProtocol))
+        svcName.getOrElse("IN-PROGRESS"),
+        svcPort.getOrElse(HttpModels.DefaultIngressServiceProtocol))
+
+  implicit val decodeUri: Decoder[Uri] = (c: HCursor) =>
+    for {
+      host <- c.as[String]
+    } yield Uri.from(host = host)
+
+  implicit val decodeCorsSupport: Decoder[CorsConfig] = (c: HCursor) =>
+    for {
+      state          <- c.downField("state").as[CorsState]
+      allowedOrigins <- c.downField("allowedOrigins").as[Set[Uri]]
+    } yield {
+      CorsConfig(state, allowedOrigins)
+    }
+
+  implicit val decodeCorsState: Decoder[CorsState] = (c: HCursor) =>
+    for {
+      maybeState <- c.as[Option[String]]
+    } yield
+      maybeState
+        .map(_.toUpperCase.trim)
+        .map {
+          case "ENABLED"  => Enabled
+          case "DISABLED" => Disabled
+          case _          => Enabled
```

Comment:
If we can only have two states, why not just check enabled and then default to disabled?